### ISSUE TITLE
[QUESTION] Suggestions on GetIceConfig Failure handling on the master side

### ISSUE THREAD

**Issue by suggestedfixes** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1012]
Because of the intermittent performance of our platform, we cannot guarantee that the master side can successfully getIceConfig one hundred percent of the time. We have switched from async getIceConfig to sync version and still observing this.

The sample app suggests to fail the connection if getIceConfig fails.
I think we have asked this question and from what we understand, the answer is no?
- Is there a way to retry the sync version of getIceConfig? For example, if getIceConfig fails, try up to two/three times instead of failing the connection? After all, it's just a rest API call.

**Comment by MushMal** [comment]
The SDK retries a number of times with progressive back-off. If the ICE refresh fails, it will give up and call the error callback which then the application can handle. The sample attempts to re-create the signaling client. Your application could do something else - like not re-create and re-try but rather monitor the network availability before creating the signaling or something along the lines.

Your application could increase the timeouts (I believe the default is 15 seconds) and the timeout per each attempt (3 seconds) if it desires but we found that if the calls fail within these times, there will not be any meaningful communication between the master and the viewer anyway.

In the latest commit we have removed the async refresh logic in the create and we are not doing automatic ICE configuration refresh. The signaling client will explicitly call ICE refresh when it's needed.
